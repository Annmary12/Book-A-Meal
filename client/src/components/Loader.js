import React from 'react';
import PropTypes from 'prop-types';
import loadingCube from '../../assets/img/loading-cube.svg';

const Loader = ({ width, message }) => (
  <div style={{ textAlign: 'center' }}>
    <img src={loadingCube} width={width} alt="loading" /><br />
    { message }
  </div>
);

Loader.propTypes = {
  width: PropTypes.number,
  message: PropTypes.string
};

Loader.defaultProps = {
  width: 40,
  message: ''
};
export default Loader;
