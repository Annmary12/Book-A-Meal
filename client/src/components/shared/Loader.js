import React from 'react';
import PropTypes from 'prop-types';
import loadingCube from '../../../assets/img/loading-cube.svg';

const Loader = ({ width }) => (
  <div style={{ textAlign: 'center' }}>
    <img src={loadingCube} width={width} alt="loading" /><br />
    loading...
  </div>
);

Loader.propTypes = {
  width: PropTypes.number
};

Loader.defaultProps = {
  width: 40
};
export default Loader;
