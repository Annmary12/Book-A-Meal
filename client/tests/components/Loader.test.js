import React from 'react';
import Loader from '../../src/components/Loader.js';

describe("Loader component", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<Loader />);

    expect(wrapper).toMatchSnapshot();
  })
})