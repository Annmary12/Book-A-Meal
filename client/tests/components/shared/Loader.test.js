import React from 'react';
import { chai, enzyme } from '../setup';
import Loader from '../../../src/components/shared/Loader.jsx';

const { render } = enzyme;

describe('Loader Component', () => {
  it('renders properly', () => {
    const tree = render(<Loader/>);
    tree.should.matchSnapshot();
  });
});
