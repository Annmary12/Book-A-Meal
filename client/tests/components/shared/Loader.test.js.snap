// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component renders properly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "style": "text-align:center",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "alt": "loading",
          "src": "[object Object]",
          "width": "40",
        },
        "children": Array [],
        "name": "img",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {},
          "children": Array [],
          "name": "br",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "data": "loading...",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "text",
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "alt": undefined,
          "src": undefined,
          "width": undefined,
        },
        "x-attribsPrefix": Object {
          "alt": undefined,
          "src": undefined,
          "width": undefined,
        },
      },
      Object {
        "attribs": Object {},
        "children": Array [],
        "name": "br",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "data": "loading...",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "text",
        },
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "alt": "loading",
            "src": "[object Object]",
            "width": "40",
          },
          "children": Array [],
          "name": "img",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "alt": undefined,
            "src": undefined,
            "width": undefined,
          },
          "x-attribsPrefix": Object {
            "alt": undefined,
            "src": undefined,
            "width": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      Object {
        "data": "loading...",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {},
          "children": Array [],
          "name": "br",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {
              "alt": "loading",
              "src": "[object Object]",
              "width": "40",
            },
            "children": Array [],
            "name": "img",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "alt": undefined,
              "src": undefined,
              "width": undefined,
            },
            "x-attribsPrefix": Object {
              "alt": undefined,
              "src": undefined,
              "width": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        "type": "text",
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
